relayer/tests/browser_test.ts

```diff
@@ -0,0 +1,152 @@
+//@ts-check
+import { AnchorProvider, BN } from "@coral-xyz/anchor";
+import {
+  Connection,
+  Keypair as SolanaKeypair,
+  PublicKey,
+} from "@solana/web3.js";
+import chai, { expect } from "chai";
+import chaiHttp from "chai-http";
+import express from "express";
+import { sign } from "tweetnacl";
+
+import {
+  Provider,
+  airdropSol,
+  User,
+  Relayer,
+  useWallet,
+  ADMIN_AUTH_KEYPAIR,
+  createTestAccounts,
+} from "@lightprotocol/zk.js";
+import sinon from "sinon";
+
+import {
+  updateMerkleTree,
+  getIndexedTransactions,
+  handleRelayRequest,
+  buildMerkleTree,
+  getLookUpTable,
+} from "../src/services";
+
+chai.use(chaiHttp);
+const app = express();
+app.use(express.json());
+app.use(express.urlencoded({ extended: false }));
+
+// Use sinon to create a stub for the middleware
+const addCorsHeadersStub = sinon
+  .stub()
+  .callsFake((_req: any, _res: any, next: any) => next());
+app.use(addCorsHeadersStub);
+
+app.post("/updatemerkletree", updateMerkleTree);
+app.get("/lookuptable", getLookUpTable);
+app.post("/relayTransaction", handleRelayRequest);
+app.get("/indexedTransactions", getIndexedTransactions);
+app.get("/getBuiltMerkletree", buildMerkleTree);
+
+describe("Browser tests", () => {
+  var RELAYER: Relayer;
+  const walletMock = useWallet(ADMIN_AUTH_KEYPAIR);
+  const connection = new Connection("http://127.0.0.1:8899", "confirmed");
+
+  before(async () => {
+    await createTestAccounts(connection);
+
+    const relayerRecipientSol = SolanaKeypair.generate().publicKey;
+    await connection.requestAirdrop(relayerRecipientSol, 2_000_000_000);
+    let relayer = SolanaKeypair.generate();
+    await airdropSol({
+      connection: connection,
+      lamports: 2_000_000_000,
+      recipientPublicKey: relayer.publicKey,
+    });
+
+    // TODO: This will only work as long as .env keys don't change
+    RELAYER = new Relayer(
+      new PublicKey("EkXDLi1APzu6oxJbg5Hnjb24kfKauJp1xCb5FAUMxf9D"),
+      new PublicKey("AV3LnV78ezsEBZebNeMPtEcH1hmvSfUBC5Xbyrzqbt44"),
+      new BN(100000),
+    );
+    await airdropSol({
+      connection: connection,
+      lamports: 2_000_000_000,
+      recipientPublicKey: walletMock.publicKey,
+    });
+  });
+  it("should fail to init node feature (anchorprovider)", async () => {
+    // should expect to throw
+    expect(() => {
+      AnchorProvider.local("http://127.0.0.1:8899", {
+        commitment: "confirmed",
+      });
+    }).to.throw("Provider local is not available on browser.");
+  });
+
+  it("should init user, shield, transfer, unshield", async () => {
+    const message =
```

Comment:
I thought we provide this abstraction in the user class. Is this necessary?
We should probably test the manually signed seed versus the one generated with the user class, right?

Reply:
not strictly necessary. added test anyway, good catch (found the internal derivation used an empty message array since the encoding didn't work properly. fixed)

Reply:
We should probably add a user instantiation test to the user tests in general.

Reply:
ah nice you  already did that